keps/sig-api-machinery/3352-aggregated-discovery/README.md

```diff
@@ -641,6 +645,25 @@ feature and upgrade/downgrade is not a problem.
 
 ### Version Skew Strategy
 
+When moving from beta to GA, we will introduce a new API group version `apidiscovery.k8s.io/v2`.
+
+All clients v1.26 to v1.29 will only request for the beta API group version `apidiscovery.k8s.io/v1beta2`.
+
+To accommodate skew between the client and server (older client and newer server), the server will serve both v2 and v2beta1 versions based on the client request headers. The API server will continue to support v2beta1 until its removal in Kubernetes v1.33.
+
+To accommodate skew between an older server and newer client, starting in v1.30,
+client-go will request for both v2 and v2beta1 by sending a list of group versions
+requested (in order from v2, v2beta1, unaggregated) and the server will return the
```

Comment:
How are these actually negotiated? If it's done using `Accept:` headers, we should use the `q` parameter to indicate preference.

Reply:
Yes it's through `Accept` headers. We've been using the parameters `g=<group>,v=<version>,as=<kind>` to denote the preference, and passing a list of acceptable gvks so the server can fallback if the first type is not supported. I added an example in the section:

```
Accept: application/json;as=APIGroupDiscoveryList;v=v2;g=apidiscovery.k8s.io,application/json;as=APIGroupDiscoveryList;v=v2beta1;g=apidiscovery.k8s.io,application/json
```

Reply:
We should say this in the KEP.